source/beamline/BioXAS/BioXASMasterValves.cpp

```diff
@@ -0,0 +1,109 @@
+#include "BioXASMasterValves.h"
+#include "beamline/BioXAS/BioXASFrontEndValves.h"
+#include "beamline/BioXAS/BioXASSideValves.h"
+#include "beamline/BioXAS/BioXASMainValves.h"
+#include "beamline/BioXAS/BioXASImagingValves.h"
+
+BioXASMasterValves::BioXASMasterValves(QObject *parent) :
+	BioXASValves("BioXASMasterValves", parent)
+{
+	// Initialize class variables.
+
+	frontEndValves_ = 0;
+	sideValves_ = 0;
+	mainValves_ = 0;
+	imagingValves_ = 0;
+}
+
+BioXASMasterValves::~BioXASMasterValves()
+{
+
+}
+
+bool BioXASMasterValves::isOpen() const
+{
+	bool open = ( isConnected() && BioXASValves::isOpen() );
```

Comment:
Implemented as is, wouldn't this code already be covered by the polymorphic attributes of `BioXASValves::isOpen()` since it already asks about the connectivity?